dapps/marketplace/src/pages/user/ProfileWizard.js

```diff
@@ -0,0 +1,493 @@
+import React, { Component, Fragment } from 'react'
+import { fbt } from 'fbt-runtime'
+import { withApollo } from 'react-apollo'
+import get from 'lodash/get'
+
+import Enum from 'utils/enum'
+import DeployIdentity from 'pages/identity/mutations/DeployIdentity'
+import withEnrolmentModal from 'pages/growth/WithEnrolmentModal'
+import { rewardsOnMobileEnabled } from 'constants/SystemInfo'
+
+import enrollmentStatusQuery from 'queries/EnrollmentStatus'
+import profileQuery from 'queries/Profile'
+
+const WizardStep = new Enum(
+  'Publish',
+  'RewardsEnroll',
+  'SetOriginProfile',
+  'SetEmail',
+  'SetPhoneNumber',
+  'VerifyYourOtherProfiles'
+)
+
+class ProfileWizard extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      uiStep: WizardStep.Publish,
+      publishChanges: false,
+      skipRewardsEnroll: false,
+      skipVerifyOtherProfiles: false,
+      userEnroledIntoRewards: false
+    }
+
+    this.EnrollButton = withEnrolmentModal('button')
+    this.componentIsMounted = false
+    setTimeout(() => this.calculateUIStep(), 1)
+  }
+
+  async componentDidMount() {
+    this.componentIsMounted = true
+    const profileResult = await this.props.client.query({
```

Comment:
We should use the `<Query>` component or higher order components instead of doing things this way